caput/memdata/fileformats.py

```diff
@@ -1,172 +1,313 @@
-        "Install with 'compression' extra_require to use bitshuffle/numcodecs compression filters."
-    )
-    compression_enabled = False
-    H5FILTER, H5_COMPRESS_LZ4 = None, None
+    H5FILTER = None
+    H5_COMPRESS_LZ4 = None
+    _compression_enabled = False
 else:
-    compression_enabled = True
-
-if compression_enabled:
-    # hdf5 parallel compression is broken before 1.13.1
-    if h5py.version.hdf5_version_tuple < (1, 13, 1):
-        import warnings
+    _compression_enabled = True
+
+# There's a bug in `hdf5` prior to version 1.13.1 which could cause
+# parallel writes to fail with certain chunk and/or compression
+# parameters. `fileformats` will still work with serial I/O, so
+# just warn the user.
+if (
+    _compression_enabled
+    and ("h5py" in BACKENDS.keys())
+    and (h5py.version.hdf5_version_tuple < (1, 13, 1))
+):
+    import warnings
+
+    warnings.warn(
+        "HDF5 parallel compression has flaws prior to version 1.13.1, and can fail "
+        f"unexpectedly. The current linked version is {h5py.version.hdf5_version_tuple}.",
+        RuntimeWarning,
+    )
 
-        warnings.warn(
-            "HDF5 parallel compression has flaws prior to version 1.13.1, and can fail "
-            f"unexpectedly. The current linked version is {h5py.version.hdf5_version_tuple}.",
-            RuntimeWarning,
-        )
 
+class FileFormat(metaclass=abc.ABCMeta):
+    """Abstract base class for file formats supported by this module.
 
-class FileFormat:
-    """Abstract base class for file formats supported by this module."""
+    Attributes
+    ----------
+    module : ModuleType
+        Python module used for file I/O.
+    compression : bool
+        Whether compression is enabled for this file format.
+    """
 
     module = None
+    compression = False
 
-    @staticmethod
-    def open(*args, **vargs):
+    @classmethod
+    def open(cls, *args, **kwargs):
         """Open a file.
 
         Not implemented in base class
+
+        Returns
+        -------
+        file_handler : Any
+            A file handler implemented by the subclass
         """
         raise NotImplementedError
 
-    @staticmethod
-    def compression_kwargs(compression=None, compression_opts=None, compressor=None):
+    @classmethod
+    def compression_kwargs(
+        cls, compression=None, compression_opts=None, compressor=None
+    ):
         """Sort compression arguments in a format expected by file format module.
 
+        Not implemented in base class.
+
         Parameters
         ----------
-        compression : str or int
+        compression : str | int | None
             Name or identifier of HDF5 compression filter.
-        compression_opts
+        compression_opts : tuple | None
             See HDF5 documentation for compression filters.
-        compressor : `numcodecs` compressor
-            As required by `zarr`.
+        compressor : :py:mod:`numcodecs` compressor
+            As required by :py:mod:`zarr`.
```

Comment:
```suggestion
            The compressor to use.
```
If we want to make it more generic.